website/tutorial/basic/05-a-simple-mutation.mdx

```diff
@@ -0,0 +1,143 @@
+
+Like before, you need to start by adding the new operation to your GraphQL schema definition.
+
+<Instruction>
+
+In `src/schema.ts`, extend the schema definition as follows:
+
+```graphql
+export const typeDefinitions = /* GraphQL */ `
+  type Query {
+    info: String!
+    feed: [Link!]!
+  }
+
+  type Mutation {
+    post(url: String!, description: String!): Link!
+  }
+
+  type Link {
+    id: ID!
+    description: String!
+    url: String!
+  }
+`
+```
+
+</Instruction>
+
+### Implementing the resolver function
+
+The next step in the process of adding a new feature to the API is to implement the resolver function for the new field.
+
+<Instruction>
+
+Next, update the `resolvers` functions to look as follows:
+
+```typescript
+type Link = {
+  id: string
+  url: string
+  description: string
+}
+
+const links: Link[] = [
+  {
+    id: 'link-0',
+    url: 'www.howtographql.com',
+    description: 'Fullstack tutorial for GraphQL',
+  },
+]
+
+const resolvers = {
+  Query: {
+    info: () => `This is the API of a Hackernews Clone`,
+    feed: () => links,
+  },
+  Mutation: {
+    post: (parent: unknown, args: { description: string; url: string }) => {
+      // 1
+      let idCount = links.length
+
+      // 2
+      const link: Link = {
+        id: `link-${idCount++}`,
+        description: args.description,
+        url: args.url,
+      }
+
+      links.push(link)
+
+      return link
+    },
+  },
+}
+```
+
+</Instruction>
+
+First off, note that you're entirely removing the `Link` object type resolver map (as explained at the very end of the last section).
+They are not needed because the GraphQL server infers what they look like.
+
+Also, here's what's going on with the numbered comments:
+
+1. You're adding a new integer variable that simply serves as a very rudimentary way to generate unique IDs for newly
+   created `Link` elements.
+1. The implementation of the `post` resolver on the `Mutation` object type first creates a new `link` object, then adds it to the existing `links`
+   array and finally returns the new `link`.
+
+Now's a good time to discuss the second argument that's passed into all resolver functions: `args`. Any guesses what
+it's used for?
+
+Correct! 💡 It carries the _arguments_ for the field that are sent – in this case the `url` and `description` of the `Link` to be
+created. We didn't need it for the `feed` and `info` resolvers before, because these `Query` object type fields don't
+specify any arguments in the schema definition.
+
+### Testing the mutation
+
+Go ahead and restart your server so you can test the new API operations. Here is a sample mutation you can send through GraphiQL:
+
+```graphql
+mutation {
+  post(url: "www.prisma.io", description: "Prisma replaces traditional ORMs") {
+    id
+  }
+}
+```
+
+The server response will look as follows:
+
+```json(nocopy)
+{
+  "data": {
+    "post": {
+      "id": "link-1"
+    }
+  }
+}
+```
+
+![Mutation response](https://i.imgur.com/MzpLiba.png)
```

Comment:
The explorer panel on the right does not display the `mutation` root field, is that expected? 👀

Reply:
screenshots will be replaced shortly